client/src/components/Connections/components/Subgraph.js

```diff
@@ -0,0 +1,237 @@
+              Ovládanie:
+              <ul>
+                <li>Ťahanie vrchola: premiestnenie vrchola v grafe</li>
+                <li>
+                  Klik na vrchol: načítať a zobraziť detailné informácie o vrchole (v boxe pod
+                  grafom)
+                </li>
+                <li>Dvojklik na vrchol: pridať do grafu nezobrazených susedov</li>
+                <li>Potrasenie vrcholom: odobrať vrchol z grafu (aj jeho výlučných susedov)</li>
+              </ul>
+            </Col>
+            <Col lg="7" md="12">
+              <div className="graph graph-legend">
+                <GraphCompnent graph={legendGraph} options={graphOptions} style={legendStyle} />
+              </div>
+            </Col>
+          </Row>
+          <div className="graph">
+            <GraphCompnent
+              graph={subgraph}
+              options={graphOptions}
+              events={{
+                select: handleSelect,
+                doubleClick: handleDoubleClick,
+                dragging: handleDrag,
+                dragEnd: handleDragEnd,
+                afterDrawing: enhanceDrawing,
+              }}
+              style={graphStyle}
+            />
+          </div>
+          {preloadNodes && subgraph.nodes.map(({id}) => <NodeLoader key={id} eid={id} />)}
+          {selectedEids.map((eid) => <InfoLoader key={eid} eid={eid} />)}
+        </div>
+      }
+    </div>
+  )
+}
+
+export default compose(
+  connect(null, {updateValue}),
+  SubgraphWrapper,
+  withHandlers({
+    handleSelect: (props: OwnProps & DispatchProps & SubgraphProps) => ({nodes}: GraphEvent) => {
+      props.updateValue(['connections', 'selectedEids'], nodes.map(getNodeEid))
+    },
+    handleDoubleClick: (props: OwnProps & DispatchProps & SubgraphProps) => ({
+      nodes,
+      pointer,
+    }: GraphEvent) => {
+      if (!nodes.length) {
+        return
+      }
+      const subgraphId = `${props.entity1.eids.join()}-${props.entity2.eids.join()}`
+      const clickedEid = getNodeEid(nodes[0]) // TODO can double click more nodes?
+
+      if (props.entityDetails[clickedEid]) {
+        const related = props.entityDetails[clickedEid].data.related
+        props.updateValue(
+          ['connections', 'subgraph', subgraphId, 'data'],
+          addNeighbours(props.subgraph, clickedEid, pointer.canvas, related)
+        )
+      }
+    },
+    handleDrag: (props: OwnProps & DispatchProps & SubgraphProps) => ({
+      nodes,
+      pointer,
+    }: GraphEvent) => {
+      if (!nodes.length) {
+        return
+      }
+      if (checkShaking(pointer.canvas)) {
+        const subgraphId = `${props.entity1.eids.join()}-${props.entity2.eids.join()}`
+        props.updateValue(
+          ['connections', 'subgraph', subgraphId, 'data'],
+          removeNodes(props.subgraph, nodes.map(getNodeEid), true)
+        )
+        props.updateValue(['connections', 'selectedEids'], [])
+      }
+    },
+    handleDragEnd: () => () => {
```

Comment:
Too much arrows :slightly_smiling_face: `() => resetGesture` would be enough. Or maybe even whole handler is redundant, just use `resetGesture` directly.